phpunit

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env php
+<?php
+
+error_reporting(-1);
+
+$PHPUNIT_VERSION = 4.8;
+$PHPUNIT_DIR = realpath(dirname($argv[0])).'/.phpunit';
+
+if (!file_exists("$PHPUNIT_DIR/phpunit-$PHPUNIT_VERSION/phpunit")) {
+    # Build a standalone phpunit without symfony/yaml
+
+    $PHPUNIT_PWD_BACKUP = getcwd();
+    mkdir($PHPUNIT_DIR);
+    chdir($PHPUNIT_DIR);
+    stream_copy_to_stream(fopen("https://github.com/sebastianbergmann/phpunit/archive/$PHPUNIT_VERSION.zip", 'rb'), fopen("$PHPUNIT_VERSION.zip", 'wb'));
+    $zip = new ZipArchive();
+    $zip->open("$PHPUNIT_VERSION.zip");
+    $zip->extractTo('.');
+    $zip->close();
+    chdir("phpunit-$PHPUNIT_VERSION");
+    passthru("composer remove --no-update symfony/yaml");
+    passthru("composer install --prefer-source --no-progress");
+    chdir($PHPUNIT_PWD_BACKUP);
+}
+
+$cmd = array_map('escapeshellarg', $argv);
+$exit = 0;
+
+if (isset($argv[1]) && 'symfony' === $argv[1]) {
+    # Find Symfony components in plain php for Windows portability
+
+    $finder = new RecursiveDirectoryIterator('src/Symfony', FilesystemIterator::KEY_AS_FILENAME | FilesystemIterator::UNIX_PATHS);
+    $finder = new RecursiveIteratorIterator($finder);
+    $finder->setMaxDepth(3);
+
+    array_shift($cmd);
+    $cmd[0] = "php $PHPUNIT_DIR/phpunit-$PHPUNIT_VERSION/phpunit".('\\' !== DIRECTORY_SEPARATOR ? " --colors=always" : '');
+    $procs = array();
+
+    foreach ($finder as $k => $v) {
+        if ('phpunit.xml.dist' === $k) {
+            $component = dirname($v->getPathname());
+
+            # Run phpunit tests in parallel
+
+            $c = escapeshellarg($component);
+
+            if ($p = proc_open(implode(' ', $cmd)." $c > $c/phpunit.stdout 2> $c/phpunit.stderr", array(), $pipes)) {
+                $procs[$component] = $p;
+            } else {
+                $exit = 1;
+                echo "\033[41mKO\033[0m $component\n\n";
```

Comment:
you should not put colors here if the terminal does not support them (by default, Windows does not)